python/ray/dataframe/dataframe.py

```diff
@@ -2595,3 +2684,8 @@ def _compute_length_and_index(dfs):
                      for j in range(lengths[i])]}
 
     return lengths, pd.DataFrame(dest_indices)
+
+
+@ray.remote
```

Comment:
Why does this need to be a remote task? I think you can just call `list(df.index)` and it will be faster.